evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -0,0 +1,110 @@
+                             "Please specify them as a dictionary with the key 'pipeline'.")
+        pipeline_params = parameters["pipeline"]
+        self.gap = pipeline_params['gap']
+        self.max_delay = pipeline_params['max_delay']
+        super().__init__(parameters, random_state)
+
+    def fit(self, X, y):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+
+        X_t = self._compute_features_during_fit(X, y)
+
+        y_shifted = y.shift(-self.gap)
+        X_t, y_shifted = keep_non_nan_rows(X_t, y_shifted)
+        self.estimator.fit(X_t, y_shifted)
+        return self
+
+    def predict(self, X, y=None, objective=None):
+        """Make predictions using selected features.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame, or np.ndarray): Data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray, None): The target training targets of length [n_samples]
+            objective (Object or string): The objective to use to make predictions
+
+        Returns:
+            pd.Series: Predicted values.
+        """
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+
+        features = self.compute_estimator_features(X, y)
+        predictions = self.estimator.predict(features.dropna(axis=0, how="any"))
+        return pad_with_nans(predictions, self.max_delay if features.isna().values.any() else 0)
+
+    def score(self, X, y, objectives):
+        """Evaluate model performance on current and additional objectives.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): Data of shape [n_samples, n_features]
+            y (pd.Series, ww.DataColumn): True labels of length [n_samples]
+            objectives (list): Non-empty list of objectives to score on
+
+        Returns:
+            dict: Ordered dictionary of objective scores
+        """
+        # Only converting X for the call to _score_all_objectives
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        y_predicted = self.predict(X, y)
+        y_shifted = y.shift(-self.gap)
```

Comment:
Yep, makes sense. So we shift the target down by `gap` rows, so that the target value at row `n` headed into the estimator comes originally from row `n + gap`. And the final `gap` rows are `nan`.